mypy/test/testcheck.py

```diff
@@ -24,87 +24,28 @@
 
 
 # List of files that contain test case descriptions.
-typecheck_files = [
-    'check-basic.test',
-    'check-union-or-syntax.test',
-    'check-callable.test',
-    'check-classes.test',
-    'check-statements.test',
-    'check-generics.test',
-    'check-dynamic-typing.test',
-    'check-inference.test',
-    'check-inference-context.test',
-    'check-kwargs.test',
-    'check-overloading.test',
-    'check-type-checks.test',
-    'check-abstract.test',
-    'check-multiple-inheritance.test',
-    'check-super.test',
-    'check-modules.test',
-    'check-typevar-values.test',
-    'check-unsupported.test',
-    'check-unreachable-code.test',
-    'check-unions.test',
-    'check-isinstance.test',
-    'check-lists.test',
-    'check-namedtuple.test',
-    'check-narrowing.test',
-    'check-typeddict.test',
-    'check-type-aliases.test',
-    'check-ignore.test',
-    'check-type-promotion.test',
-    'check-semanal-error.test',
-    'check-flags.test',
-    'check-incremental.test',
-    'check-serialize.test',
-    'check-bound.test',
-    'check-optional.test',
-    'check-fastparse.test',
-    'check-warnings.test',
-    'check-async-await.test',
-    'check-newtype.test',
-    'check-class-namedtuple.test',
-    'check-selftype.test',
-    'check-python2.test',
-    'check-columns.test',
-    'check-functions.test',
-    'check-tuples.test',
-    'check-expressions.test',
-    'check-generic-subtyping.test',
-    'check-varargs.test',
-    'check-newsyntax.test',
-    'check-protocols.test',
-    'check-underscores.test',
-    'check-classvar.test',
-    'check-enum.test',
-    'check-incomplete-fixture.test',
-    'check-custom-plugin.test',
-    'check-default-plugin.test',
-    'check-attr.test',
-    'check-ctypes.test',
-    'check-dataclasses.test',
-    'check-final.test',
-    'check-redefine.test',
-    'check-literal.test',
-    'check-newsemanal.test',
-    'check-inline-config.test',
-    'check-reports.test',
-    'check-errorcodes.test',
-    'check-annotated.test',
-    'check-parameter-specification.test',
-    'check-generic-alias.test',
-    'check-typeguard.test',
-]
+typecheck_files = []
+
+# Define the path where the check files are present.
+if sys.platform == 'win32':
+    path2loc = os.path.abspath(__file__).split('mypy\\test')[0] + 'test-data\\unit'
+else:
+    path2loc = os.path.join(os.path.abspath(__file__).split('mypy/test')[0], 'test-data/unit')
+
+for check_files in os.listdir(path2loc):
+    # Append only if it has the pattern ( check-*-*.test)
+    if re.match(r'check-\w+-?\w+?-?\w+?.test', check_files):
+        typecheck_files.append(check_files)
```

Comment:
Can you move this and the above logic to a helper module, so that this can be reused in other places easily?

For example, we could have something like this in this file:

```
typecheck_files = expand_test_files('test-data/unit/check-*.test')
```

You can use `glob.glob` to expand patterns instead of `re`, since regular expressions are more error-prone as they require escaping, etc.